### PR TITLE
Introduce the mdelay function.

### PR DESCRIPTION
**Pull request by MichalBlk**
The `mdelay` function waits for a given amount of ms. It's devoted for use inside device drivers and any other field which requires accurate waiting at ms level.

### CODE REVIEW

File: sys/kern/time.c [hunk 1]
```diff
@@ -47,4 +49,15 @@ time_t tm2sec(tm_t *t) {
...
+
+static void mdelay_timeout(__unused void *arg) {
+  /* Nothing to do here. */
+}
+
+void mdelay(useconds_t ms) {
+  callout_t handle;
+  bzero(&handle, sizeof(callout_t));
+  callout_setup_relative(&handle, ms, mdelay_timeout, NULL);
+  callout_drain(&handle);
```
Comment (wiklam) [inline sys/kern/time.c:62]: As far as I know, `mdelay` is busy waiting, and you implemented probably `msleep`.

File: sys/kern/time.c [hunk 2]
```diff
@@ -47,4 +49,15 @@ time_t tm2sec(tm_t *t) {
...
+}
```
Comment (pj1031999) [inline sys/kern/time.c:63]: It's not clear for me. Argument for `mdelay` is ms (millisecond?) but `useconds_t` looks like microseconds. On the other hand second argument of `callout_setup_relative` is a `systime_t`. Could you explain?

Comment (mohrcore) [reply sys/kern/time.c:63]: I'm having similar thoughts on this one. I'm be waiting for an explanation. Other than that, the logic seems fine, but I can't accept it unless you clarify what's going on here.

**Comment by cahirwpz** [comment]
@MichalBlk Since we don't have clear use case for this function, the code has rotten and is not eligible for merging anymore. Please close this PR.

**Pull request closed without merging**